lib/valueflow.cpp

```diff
@@ -8318,7 +8318,7 @@ static void valueFlowContainerSize(TokenList* tokenlist,
 
     // after assignment
     for (const Scope *functionScope : symboldatabase->functionScopes) {
-        for (Token* tok = const_cast<Token*>(functionScope->bodyStart); tok != functionScope->bodyEnd; tok = tok->next()) {
+        for (auto* tok = const_cast<Token*>(functionScope->bodyStart); tok != functionScope->bodyEnd; tok = tok->next()) {
```

Comment:
I am not sure if this makes the code more robust.

You just save 1 byte in the type name. If `Token*` is written that is more explicit about what you want than `auto*`.. and if rhs would mismatch I envision there will be compiler errors that you don't get with a `auto*` type.

Reply:
In this case there's a cast which already specifies the resulting type so it is duplicated. That's the only pattern I have touched so far since it makes sense IMO.

The check also has a threshold for the length of the type it should report on.

Reply:
> That's the only pattern I have touched so far since it makes sense IMO.

thanks.

> The check also has a threshold for the length of the type it should report on.

Maybe that can be extended then.. how about something like 15 or so. It's quite aggressive when it just saves 1 byte at least.

Reply:
> Maybe that can be extended then.. how about something like 15 or so. It's quite aggressive when it just saves 1 byte at least.

At least for me this is not about saving code but to ease making changes so you don't have to change types multiple times with maintaining readability of the code. In case of using it for an iterator you might not easily found out what type(s) are behind it. Especially if the variable is out-of-view/scope.

If you are using an IDE with refactoring and tooltips that would not be an issue in either case.

Reply:
> At least for me this is not about saving code but to ease making changes so you don't have to change types multiple times with maintaining readability of the code.

I want it to be robust. How much manual editing you must do is not important.

If the type is changed automatically "behind your back" you can get bugs.

But manual editing can also lead to mistakes.

> If you are using an IDE with refactoring and tooltips that would not be an issue in either case.

Yes but imho such IDE is not very usable. I like lightweight and quick IDEs.

Reply:
> I want it to be robust. How much manual editing you must do is not important.

If the type is specified explicitly at least one in the line I would consider it "robust" as well as readable.

Reply:
I would like that the threshold is increased to make this less aggressive. Can we meet somewhere in the middle.. 10? I appreciate that you did not replace with auto everywhere where it's technically possible.